### PR TITLE
Cpu any simd

### PR DESCRIPTION
**Pull request by gvidinski**

### CODE REVIEW

File: period_search_optimization_simd/period_search/CpuInfoArm.cpp [hunk 1]
```diff
@@ -8,22 +8,75 @@
 #include "CalcStrategyNone.hpp"
 #include "CalcStrategyAsimd.hpp"
 
+#if defined(__linux__) && (defined(__arm__) || defined(_M_ARM) || defined(__aarch64__) || defined(_M_ARM64))
 #if defined(__linux__) && (defined(__arm__) || defined(_M_ARM) || defined(__aarch64__) || defined(_M_ARM64))
```
Comment (ahorek) [inline period_search_optimization_simd/period_search/CpuInfoArm.cpp:12]: conflict

File: period_search_optimization_simd/period_search/CpuInfoArm.cpp [hunk 2]
```diff
@@ -71,9 +124,11 @@ void SetOptimizationStrategy(SIMDEnum useOptimization)
 {
 	switch (useOptimization)
 	{
+#if !(defined(__linux__) && (defined(__aarch64__) || defined(_M_ARM64)))
```
Comment (ahorek) [inline period_search_optimization_simd/period_search/CpuInfoArm.cpp:127]: never use asimd on 64bit linux? this looks wrong or at least not in the right place...

File: period_search_optimization_simd/period_search/CpuInfoArm.cpp [hunk 3]
```diff
@@ -8,22 +8,75 @@
...
 void GetSupportedSIMDs()
 {
 	#if defined(__linux__)
 	  #if (defined(__aarch64__) || defined(_M_ARM64))
 	    uint64_t hwcap = getauxval(AT_HWCAP);
-        CPUopt.hasASIMD = hwcap & HWCAP_ASIMD;
+		// NOTE: For debug purposes:
+		// DetectArmv8CpuFeatures(hwcap);
+
+        CPUopt.hasASIMD = hwcap & HWCAP_PMULL & HWCAP_ASIMD;
```
Comment (ahorek) [inline period_search_optimization_simd/period_search/CpuInfoArm.cpp:79]: is PMULL a remnant from testing? I don't get why it should be necessary....

**Comment by gvidinski** [comment]
You're right. This is already fixed and will be there within the next PR.

**Pull request merged**